app/interactors/validate_user.rb

```diff
@@ -0,0 +1,11 @@
+class ValidateUser
+  include Interactor
+
+  def call
+    context.user = User.find_by(email: context.email.downcase)
+    user = context.user
+    password = context.password
+    errors = context.user.errors
+    context.fail!(errors: errors) unless user && user.authenticate(password)
```

Comment:
@Enriikke I needed to do something to get this line shorter (I hate you, rubocop). A multi-line conditional raises another style issue. Suggestions?